src/Compilers/CSharp/Portable/CSharpExtensions.cs

```diff
@@ -1220,6 +1220,15 @@ public static ISymbol GetDeclaredSymbol(this SemanticModel semanticModel, Declar
         }
 
         /// <summary>
+        /// Given a declaration pattern syntax, get the corresponding symbol.
+        /// </summary>
+        public static ISymbol GetDeclaredSymbol(this SemanticModel semanticModel, LetStatementSyntax declarationSyntax, CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
I don't think this API should be here.

Symbols are not declared "by" the let statement, any more than they are declared by an "is" expression with a pattern on the right-hand-side. Such symbols (zero, one, or many) are instead declared by the patterns appearing *inside* the let statement.